libethereum/ExtVM.cpp

```diff
@@ -134,3 +134,22 @@ void ExtVM::suicide(Address _a)
 	m_s.subBalance(myAddress, m_s.balance(myAddress));
 	ExtVMFace::suicide(_a);
 }
+
+h256 ExtVM::blockHash(u256 _number)
+{
+	if (envInfo().number() < m_sealEngine.chainParams().u256Param("metropolisForkBlock") + 256)
```

Comment:
Oh, I see, I was assuming that the contract itself (the new version of it) will return 0 if requested block is out of bounds.
Yeah, anyway it will makes sense to avoid making a call in this case, I'll change it.
Thanks for noticing this!